src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetWidget.java

```diff
@@ -694,8 +694,10 @@ public void adaptToFileType(TextFileType fileType)
       
       sourceOnSave_.setVisible(canSourceOnSave);
       srcOnSaveLabel_.setVisible(canSourceOnSave);
-      if (fileType.isRd() || fileType.isJS() || canPreviewFromR || fileType.isSql() )
+      if (fileType.isRd() || fileType.isJS() || canPreviewFromR || fileType.isSql())
          srcOnSaveLabel_.setText(fileType.getPreviewButtonText() + " on Save");
+      else if (hasCustomSource())
+         srcOnSaveLabel_.setText("Function on Save");
```

Comment:
Not a fan of 'Function on Save'; I think keeping it as 'Source on Save' is fine.

Reply:
Yeah, lets keep it the way it is.